numpy/core/src/umath/ufunc_object.c

```diff
@@ -2457,7 +2457,7 @@ PyUFunc_GeneralizedFunctionInternal(PyUFuncObject *ufunc, PyArrayObject **op,
     /* Fill in any allocated outputs */
     {
         PyArrayObject **operands = NpyIter_GetOperandArray(iter);
-        for (i = 0; i < nop; ++i) {
+        for (i = nin; i < nop; ++i) {
```

Comment:
No point in iterating through all, `op[i]` cannot be NULL unless `i >= nin`.

Reply:
Frankly, this could be moved into the later loop, but for clarity, I wasn't sure that would actually be better...